File: frontend/src/types/Poll.ts
declare namespace BeezlePolls {
    interface PollVoter {
        poll_id: string;
        handle: string;
        value: string;
    }
    interface Poll {
        title: string;
        options: Array<string>;
        poll_id: string;
        post_id: string;
        voters: Array<PollVoter>;
        expiry_date: {
            $date: {
                $numberLong: string;
            };
        };
    }
}

File: backend/src/mongoose/structures/poll.rs
use bson::{serde_helpers::chrono_datetime_as_bson_datetime, Array};
use mongodb::bson::oid::ObjectId;
use serde::{Deserialize, Serialize};


#[derive(Serialize, Deserialize, Debug)]
pub struct Poll {
    #[serde(rename = "_id", skip_serializing_if = "Option::is_none")]
    pub id: Option<ObjectId>,
    pub poll_id: String,
    pub title: String,
    pub options: Array,
    pub post_id: String,
    #[serde(with = "chrono_datetime_as_bson_datetime")]
    pub expiry_date: chrono::DateTime<chrono::Utc>,
    pub over: bool
}

#[derive(Serialize, Deserialize, Debug)]
pub struct PollVoter {
    #[serde(rename = "_id", skip_serializing_if = "Option::is_none")]
    pub id: Option<ObjectId>,
    pub poll_id: String,
    pub handle: String,
    pub value: String
}

